From 7c506f574baecf6ff05368a8340d650f66468381 Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Fri, 3 Apr 2026 12:09:21 +0200
Subject: [PATCH 76/91] dt-bindings: phy: amlogic: usb3: Add compatible for
 Amlogic T7 SoC

On the G12A, a single combo PHY handles both USB3 and PCIe.
The Amlogic T7 splits this into two dedicated PHYs,
one for USB3 and one for PCIe, but the USB3 PHY register layout
remains compatible with the G12A combo PHY.
Add amlogic,t7-usb3-phy with a fallback to amlogic,g12a-usb3-pcie-phy.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
---
 .../bindings/phy/amlogic,g12a-usb3-pcie-phy.yaml         | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/phy/amlogic,g12a-usb3-pcie-phy.yaml b/Documentation/devicetree/bindings/phy/amlogic,g12a-usb3-pcie-phy.yaml
index 1a5a12adb72b..16c72cb80c47 100644
--- a/Documentation/devicetree/bindings/phy/amlogic,g12a-usb3-pcie-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/amlogic,g12a-usb3-pcie-phy.yaml
@@ -12,8 +12,13 @@ maintainers:
 
 properties:
   compatible:
-    enum:
-      - amlogic,g12a-usb3-pcie-phy
+    oneOf:
+      - items:
+          - enum:
+              - amlogic,t7-usb3-phy
+          - const: amlogic,g12a-usb3-pcie-phy
+      - enum:
+          - amlogic,g12a-usb3-pcie-phy
 
   reg:
     maxItems: 1
-- 
2.53.0

